server/task/task-memory/src/main/java/org/apache/james/task/SerialTaskManagerWorker.java

```diff
@@ -40,22 +34,23 @@
 import reactor.core.Disposable;
 import reactor.core.publisher.Flux;
 import reactor.core.publisher.Mono;
+import reactor.core.scheduler.Scheduler;
 import reactor.core.scheduler.Schedulers;
 import reactor.util.function.Tuple2;
 import reactor.util.function.Tuples;
 
 public class SerialTaskManagerWorker implements TaskManagerWorker {
-
     private static final Logger LOGGER = LoggerFactory.getLogger(SerialTaskManagerWorker.class);
-    private final ExecutorService taskExecutor;
+
+    private final Scheduler taskExecutor;
     private final Listener listener;
-    private final AtomicReference<Tuple2<TaskId, Future<?>>> runningTask;
+    private final AtomicReference<Tuple2<TaskId, Disposable>> runningTask;
     private final Set<TaskId> cancelledTasks;
     private final Duration pollingInterval;
 
     public SerialTaskManagerWorker(Listener listener, Duration pollingInterval) {
         this.pollingInterval = pollingInterval;
-        this.taskExecutor = Executors.newSingleThreadExecutor(NamedThreadFactory.withName("task executor"));
+        this.taskExecutor = Schedulers.newBoundedElastic(1, 1, "task executor");
```

Comment:
why did you choose that specific scheduler and why did you limit the scheduler queue?

Reply:
I wanted to use `single` but it rejected calls to `.block` which is a no-go here.

A single thread bounded ealstic scheduler have no such limitations.

Reply:
you answered only half of the question: why a limit of the queue? BTW, you can create a Scheduler from an Executor if needed.

Reply:
> BTW, you can create a Scheduler from an Executor if needed.

Let's do it then